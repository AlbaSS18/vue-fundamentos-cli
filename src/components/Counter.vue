<template>
  <!-- Para crear el componente, se puede poner Vue, y por defecto se crea la estructura-->
  <h2>{{customTitle}} {{start}}</h2>
  <p>{{counter}} <sup>2</sup> = {{ squareCounter }}</p>
  <p data-testid="counter">{{counter}}</p>

  <div class="buttons-container">
      <button @click="increase">+1</button>
      <!-- v-on: es sinÃ³nimo de @ -->
      <button @click="decrease">-1</button>
  </div>
</template>

<script>
export default {
    props: {
        title: String,
        start: {
            type: Number,
            required: false,
            default: 200,
            validator (value) {
                return value > 100;
            }
        }
    },
    // name: 'Patito' // Se puede poner nombre a los componentes para identificarlos
    data() {
        return {
            counter : this.start
        }
    },
    methods: {
        getSquare(){
            return this.counter * this.counter;
        },
        increase() {
            this.counter ++; // No hace falta poner return
        },
        decrease(){
            this.counter --;
        }
    },
    computed: {
        squareCounter(){
            return this.counter * this.counter;
        },
        customTitle(){
            return this.title || 'Counter';
        }
    }
}
</script>

<style>

button {
    background-color: #64BB87;
    border-radius: 5px;
    border: 1px solid white;
    color: white;
    cursor: pointer;
    margin: 0 5px;
    padding: 5px 15px;
    transition: 0.3s ease-in-out;
}

button:hover {
    background-color: #5aa67b;
    transition: 0.3s ease-in-out;
}


</style>