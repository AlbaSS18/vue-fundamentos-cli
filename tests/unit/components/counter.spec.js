
import {shallowMount, mount} from '@vue/test-utils'
import Counter from '@/components/Counter'

/**
 * MOUNT: monta el componente y los subcomponentes.
 * SHALLOWMOUNT: monta el componente actual.
 */
describe('Counter component', () => {
    
    test('debe de hacer match con el snapshot', () => {
        const wrapper = shallowMount(Counter);

        expect(wrapper.html()).toMatchSnapshot();
        // Copia f√≠sica del componente
    })
})