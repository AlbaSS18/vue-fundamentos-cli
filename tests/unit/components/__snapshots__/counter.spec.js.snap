// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component debe de hacer match con el snapshot 1`] = `
<!-- Para crear el componente, se puede poner Vue, y por defecto se crea la estructura-->
<h2>Counter 200</h2>
<p>200 <sup>2</sup> = 40000</p>
<div class="buttons-container"><button>+1</button><!-- v-on: es sinÃ³nimo de @ --><button>-1</button></div>
`;
